### PR TITLE
My Site Dashboard: add the initial structure for the posts card

### PR DESCRIPTION
**Pull request by leandroalonso**
Part of #17697

This PR adds the initial structure to show 3 draft posts. It's not possible to interact with them neither the syncing is done, this will be tackled in future PRs.

I also want to decouple the `ViewController` and the `ViewModel`. I'll do it in a follow-up PR but please if you have any feedback go ahead!

### To test

1. Copy and paste [this gist](https://gist.github.com/leandroalonso/ba1c99305807e19839ba7b11a48c1a17) into `BlogViewController.m`
2. Run the app
3. Make sure that the first 3 draft posts appear
4. If they're not appearing, go to posts > draft
5. Close and reopen the app*

*Syncing is not available yet, so the posts need to be on the database.

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
I plan to add unit tests for the view model in an upcoming PR.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+}
+
+private extension PostsCardViewController {
+    func configureView() {
+        view.addSubview(postsTableView)
+        postsTableView.translatesAutoresizingMaskIntoConstraints = false
+        postsTableView.pinSubviewToAllEdges(view)
+        let postCompactCellNib = UINib(nibName: "PostCompactCell", bundle: Bundle.main)
+        postsTableView.register(postCompactCellNib, forCellReuseIdentifier: "PostCompactCellIdentifier")
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]: [nit] If we make `PostCompactCell` conform to `NibReusable` we can then do `PostCompactCell.defaultNib` and `PostCompactCell.defaultReuseID`

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        postsTableView.dataSource = viewModel
+        viewModel.refresh()
+    }
+}
+
+private extension PostsCardViewController {
+    func configureView() {
+        view.addSubview(postsTableView)
+        postsTableView.translatesAutoresizingMaskIntoConstraints = false
+        postsTableView.pinSubviewToAllEdges(view)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]:
```suggestion
        view.pinSubviewToAllEdges(postsTableView)
```

[nit] This technically has the same result, but the param should be the subview being pinned to the superview
```
    /// Adds constraints that pin a subview to self with zero insets.
    ///
    /// - Parameter subview: a subview to be pinned to self.
    @objc public func pinSubviewToAllEdges(_ subview: UIView) {
        pinSubviewToAllEdges(subview, insets: .zero)
    }
```

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @objc var blog: Blog?
+
+    private let postsTableView = IntrinsicTableView()
+
+    private var viewModel: PostsCardViewModel!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureView()
+        viewModel = PostsCardViewModel(tableView: postsTableView, blog: blog!)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]: What do you think of passing a non-optional blog in the init method so we don't have to implicitly unwrap the blog object?

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+        fetchedResultsController.delegate = self
+    }
+
+    func fetchRequest() -> NSFetchRequest<Post> {
+        let fetchRequest = NSFetchRequest<Post>(entityName: String(describing: Post.self))
+        fetchRequest.predicate = predicateForFetchRequest()
+        fetchRequest.sortDescriptors = sortDescriptorsForFetchRequest()
+        fetchRequest.fetchBatchSize = 3
+        fetchRequest.fetchLimit = 3
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewModel.swift]: What do you think of defining a constant for the batch size / fetch limit?

**Review by momo-ozawa — COMMENTED** [review]
Works as described, left some comments.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]: Done in 963aec3e6d55c353f9065ae9eb0eb03ddded26e2

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]: Done in b44161ae2c73cf8d720ab4197c535575bce1f2e3

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift (hunk 3 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewController.swift]: Done in 8393babaf7e17796f9bca2242715377d80aef55e

File: WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewModel.swift (hunk 4 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/PostsCardViewModel.swift]: Done 26760b16bbdf4470da167af91d2d765095aa7b23

**Comment by leandroalonso** [comment]
@momo-ozawa addressed all your comments, thanks for the suggestions!

File: WordPress/Classes/ViewRelated/Post/PostCompactCell.swift [hunk 5]
```diff
@@ -204,3 +204,7 @@ extension PostCompactCell: GhostableView {
         static let opacity: Float = 0.5
     }
 }
+
+extension PostCompactCell: NibReusable {
+    static var defaultReuseID = "PostCompactCellIdentifier"
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Post/PostCompactCell.swift]: FYI NibReusable has a default implementation for defaultReuseID so you don't need to provide an implementation :)

**Review by momo-ozawa — APPROVED** [review]
Looks good! Thanks for the changes, left 1 small comment

File: WordPress/Classes/ViewRelated/Post/PostCompactCell.swift (hunk 5 above)
Comment (leandroalonso) [reply WordPress/Classes/ViewRelated/Post/PostCompactCell.swift]: Fixed!

**Comment by leandroalonso** [comment]
@momo-ozawa I'll wait for your PR to be merged so I can put these files in the correct place.

**Pull request merged**